File: transaction-service/src/main/java/ua/com/foxminded/restClient/exceptions/CurrencyNotFoundException.java
package ua.com.foxminded.restClient.exceptions;

public class CurrencyNotFoundException extends RuntimeException {

    static private final String message = "Currency rate between %s and %s not found.";

    public CurrencyNotFoundException() {
        super();
    }

    public CurrencyNotFoundException(String s, Throwable cause) {
        super(s, cause);
    }

    public CurrencyNotFoundException(Throwable cause) {
        super(cause);
    }

    public CurrencyNotFoundException(String s) {
        super(s);
    }

    public CurrencyNotFoundException(String firstCurrency, String secondCurrency) {
        super(message.formatted(firstCurrency, secondCurrency));
    }
}

File: transaction-service/src/main/java/ua/com/foxminded/restClient/exceptions/CurrencyNotFoundException.kt
package ua.com.foxminded.restClient.exceptions

class CurrencyNotFoundException : RuntimeException {
    constructor() : super()
    constructor(s: String?, cause: Throwable?) : super(s, cause)
    constructor(cause: Throwable?) : super(cause)
    constructor(s: String?) : super(s)
    constructor(firstCurrency: String?, secondCurrency: String?) : super(
        Companion.message.format(
            firstCurrency,
            secondCurrency
        )
    )

    companion object {
        private const val message = "Currency rate between %s and %s not found."
    }
}
